### PR TITLE
Fix Button's `sizeThatFits` calculation

### PR DESCRIPTION
**Pull request by huwilkes**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

A client was having an issue where their button would break into two lines. The issue seems to be coming from `proposedTitleLabelWidth`, so we're removing that from the `sizeThatFits` calculation for now.

### Verification

Tested in their app, as well as in our test app for iOS 14, 15, and 16.

<details>
<summary>Visual Verification</summary>

| iOS Version | Before | After |
|---|---|---|
| 14 | Button_14_Before | Button_14_After |
| 15 | Button_15_Before | Button_15_After |
| 16 | Button_16_Before | Button_16_After |
</details>

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1692)

### CODE REVIEW

**Comment by harrieshin** [comment]
why then have updateProposedTitleLabelWidth() at all?

**Comment by huwilkes** [comment]
> why then have updateProposedTitleLabelWidth() at all?

Good question. I was going with the smallest possible change in 0.14 and 0.15, but was going to work on a more comprehensive change into main (hopefully before 16) and just remove it if possible.

**Comment by harrieshin** [comment]
> > why then have updateProposedTitleLabelWidth() at all?
> 
> Good question. I was going with the smallest possible change in 0.14 and 0.15, but was going to work on a more comprehensive change into main (hopefully before 16) and just remove it if possible.

what the comprehensive change mean? there is still side effect of updateProposedTitleLabelWidth is used for?

**Comment by huwilkes** [comment]
> > > why then have updateProposedTitleLabelWidth() at all?
> > 
> > 
> > Good question. I was going with the smallest possible change in 0.14 and 0.15, but was going to work on a more comprehensive change into main (hopefully before 16) and just remove it if possible.
> 
> what the comprehensive change mean? there is still side effect of updateProposedTitleLabelWidth is used for?

The comprehensive change would include actually removing `proposedTitleLabelWidth`, and seeing if I could just remove the custom `intrinsicContentSize` and `sizeThatFits`. I tried removing `proposedTitleLabelWidth`, but the resizing issue repos if you just rip it out all the way.

**Comment by huwilkes** [comment]
> > > > why then have updateProposedTitleLabelWidth() at all?
> > > 
> > > 
> > > Good question. I was going with the smallest possible change in 0.14 and 0.15, but was going to work on a more comprehensive change into main (hopefully before 16) and just remove it if possible.
> > 
> > 
> > what the comprehensive change mean? there is still side effect of updateProposedTitleLabelWidth is used for?
> 
> The comprehensive change would include actually removing `proposedTitleLabelWidth`, and seeing if I could just remove the custom `intrinsicContentSize` and `sizeThatFits`. I tried removing `proposedTitleLabelWidth`, but the resizing issue repos if you just rip it out all the way.

Switching to `invalidateIntrinsicContentSize` works, though.

File: ios/FluentUI/Button/Button.swift [hunk 1]
```diff
@@ -154,7 +154,7 @@ open class Button: UIButton, TokenizedControlInternal {
     open override func layoutSubviews() {
```
Comment (harrieshin) [inline ios/FluentUI/Button/Button.swift:154]: talked offline. removing this will work .

**Review by harrieshin — APPROVED** [review]

**Review by mischreiber — APPROVED** [review]

**Comment by ryan-w** [comment]
Hey @harrieshin and @mischreiber, I believe the `proposedTitleLabelWidth` was there to work around a bug with the `UIButton` and multi-line text. You can see this when the text size is increased and the button's text wraps.

This is screenshot before this update (from v0.16.0):

Before-Multi-line-button-v0-16-0

And here's a screenshot after:

After-Multi-line-button-v-Main

**Pull request merged**